torch/csrc/jit/pickler.cpp

```diff
@@ -197,6 +187,20 @@ void Pickler::addIValueImpl(const IValue& ivalue) {
             addIValue(item);
           }
         });
+  } else if (ivalue.isObject()) {
+    auto obj = ivalue.toObject();
+    auto type = obj->type();
+    pushGlobal(type->qualifier(), type->basename());
+    push<OpCode>(OpCode::EMPTY_TUPLE);
+    push<OpCode>(OpCode::NEWOBJ);
+    push<OpCode>(OpCode::EMPTY_DICT);
+    push<OpCode>(OpCode::MARK);
+    for (size_t i = 0, n = type->numAttributes(); i < n; ++i) {
```

Comment:
We should check if `__getstate__` is present and use that if so instead of the state dict, but that's fine to leave for a later PR